src/compiler/parser.ts

```diff
@@ -2973,6 +2981,44 @@ namespace ts {
             return parseParenthesizedArrowFunctionExpressionHead(/*allowAmbiguity*/ false);
         }
 
+        function tryParseUnparenthesizedAsyncArrowFunctionExpressionWorker(): ArrowFunction {
+            const isUnParenthesizedAsyncArrowFunction = lookAhead(isUnParenthesizedAsyncArrowFunctionWorker);
+            if (isUnParenthesizedAsyncArrowFunction === Tristate.True) {
+                const node = <ArrowFunction>createNode(SyntaxKind.ArrowFunction);

```

Comment:
Yea. I am thinking about repurpose the `parseSimpleArrowFunctionExpression ` and pass in whether it is async arrow-function or not.